src/js/parsers/rssparser.ts

```diff
@@ -31,17 +38,25 @@ export default function parseRss(dat) {
 }
 
 // Translate RSS item to playlist item.
-function parseItem(obj) {
-    const item = {};
+function parseItem(obj: PageNode): PlaylistItemType {
+    const item: {
+        file?: string;
+        title?: string;
+        mediaid?: string;
+        date?: string;
+        description?: string;
+        link?: string;
+        tags?: string;
+    } = {};
     for (let i = 0; i < obj.childNodes.length; i++) {
-        const node = obj.childNodes[i];
+        const node = obj.childNodes[i] as PageNode;
```

Comment:
What does as PageNode do here?

Reply:
TS knows that `obj` is a dom node and `childNodes` is a pre-typed `ChildNode` unfortunately for what I presume are compatibility reasons with older browsers, we look for information on these nodes that TS doesn't think is present on a `ChildNode` type (such as `.text`) - so what I'm doing here is telling TS to treat these `ChildNodes` all as `PageNode` since `PageNode` is a union type of the `ChildNode` and some of those additional compat properties

Reply:
I'm unclear on how safe or unsafe this type of operation is but i assume we'll be coming across it a lot, so would be good to discuss as a team

Reply:
It could lead to a lot of transgressions of open/close principle, but I'm down to discuss.

Reply:
Also lets you totally bypass required properties:
![image](https://user-images.githubusercontent.com/2903955/74048553-6f5a3800-49a0-11ea-83dd-55e4a238e975.png)

Usually in TS union types can be done with this pattern safely:

```
interface IStudent {
    id: string;
    age: number;
}

interface IWorker {
    companyId: string;
}

type IUnionType = IStudent | IWorker;
```